sdk/cosmos/azure_data_cosmos/tests/cosmos_items.rs

```diff
@@ -405,3 +407,341 @@ pub async fn item_null_partition_key(context: TestContext) -> Result<(), Box<dyn
     account.cleanup().await?;
     Ok(())
 }
+
+#[recorded::test]
+pub async fn item_replace_if_match_etag(context: TestContext) -> Result<(), Box<dyn Error>> {
+    let account = TestAccount::from_env(context, None).await?;
+    let cosmos_client = account.connect_with_key(None)?;
+    let container_client = create_container(&account, &cosmos_client).await?;
+
+    //Create an item
+    let mut item = TestItem {
+        id: "Item1".into(),
+        partition_key: Some("Partition1".into()),
+        value: 42,
+        nested: NestedItem {
+            nested_value: "Nested".into(),
+        },
+        bool_value: true,
+    };
+
+    let response = container_client
+        .create_item("Partition1", &item, None)
+        .await?;
+
+    //Store Etag from response
+    const ETAG: HeaderName = HeaderName::from_static("etag");
+    let etag = response
+        .headers()
+        .get_str(&ETAG)
+        .ok()
+        .map(|s| Etag::from(s.to_string()));
+
+    //Replace item with correct Etag
+    item.value = 24;
+    item.nested.nested_value = "Updated".into();
+
+    //could change response name to replaced item for better understanding
+    let response = container_client
+        .replace_item(
+            "Partition1",
+            "Item1",
+            &item,
+            Some(ItemOptions {
+                if_match_etag: etag,
+                enable_content_response_on_write: false,
+                ..Default::default()
+            }),
+        )
+        .await?;
+
+    let body = response.into_raw_body().collect_string().await?;
+    assert_eq!("", body);
+
+    //Replace item with incorrect Etag
+    item.value = 52;
+    item.nested.nested_value = "UpdatedAgain".into();
+
+    let response = container_client
+        .replace_item(
+            "Partition1",
+            "Item1",
+            &item,
+            Some(ItemOptions {
+                if_match_etag: Some("incorrectEtag".into()),
+                enable_content_response_on_write: true,
+                ..Default::default()
+            }),
+        )
+        .await;
+
+    match response {
+        Ok(_) => {
+            return Err(
+                "expected a 412 Precondition Failed error when using an incorrect ETag".into(),
+            );
+        }
+        Err(err) => {
+            assert_eq!(
+                Some(azure_core::http::StatusCode::PreconditionFailed),
+                err.http_status()
+            );
+        }
+    }
+
+    account.cleanup().await?;
+    Ok(())
+}
+
+#[recorded::test]
+pub async fn item_upsert_if_match_etag(context: TestContext) -> Result<(), Box<dyn Error>> {
```

Comment:
Most of the feedback from the above test applies to this one as well, so I won't repeat it all here :)